ignite/cmd/chain.go

```diff
@@ -60,13 +72,79 @@ chain.
 		Args:    cobra.ExactArgs(1),
 	}
 
-	c.AddCommand(
-		NewChainServe(),
-		NewChainBuild(),
-		NewChainInit(),
-		NewChainFaucet(),
-		NewChainSimulate(),
-	)
+	c.AddCommand(addConfigMigrationVerifier(NewChainServe()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainBuild()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainInit()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainFaucet()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainSimulate()))
 
 	return c
 }
+
+func addConfigMigrationVerifier(cmd *cobra.Command) *cobra.Command {
+	cmd.Flags().AddFlagSet(flagSetConfig())
+	cmd.Flags().AddFlagSet(flagSetYes())
+
+	preRunFun := cmd.PreRunE
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) (err error) {
+		if preRunFun != nil {
+			if err = preRunFun(cmd, args); err != nil {
+				return err
+			}
+		}
+
+		configPath := getConfig(cmd)
+		if configPath == "" {
+			appPath := flagGetPath(cmd)
+
+			if configPath, err = chainconfig.LocateDefault(appPath); err != nil {
+				return err
+			}
+		}
+
+		rawCfg, err := ioutil.ReadFile(configPath)
+		if err != nil {
+			return err
+		}
+
+		version, err := chainconfig.ReadConfigVersion(bytes.NewReader(rawCfg))
+		if err != nil {
+			return err
+		}
+
+		// Config files with older versions must be migrated to the latest before executing the command
+		if version != chainconfig.LatestVersion {
+			// TODO: When "--yes" flag is present print a warning message to inform of config migration (cliui)?
+			// Confirm before overwritting the config file
+			if !getYes(cmd) {
+				confirmed := false
+				prompt := &survey.Confirm{
+					Message: fmt.Sprintf(migrateMsg, version, chainconfig.LatestVersion),
+				}
+
+				if err := survey.AskOne(prompt, &confirmed); err != nil {
```

Comment:
Is there a reason behind using `survey` over our internal `cliui.Ask()` ?

Reply:
Maybe because at the time of the contribution it was not know. I was actually planning to add `cliui` later on but given your feedback I have done it in f1ae0eb4697957b1d3c20c3d78e189c1f16a1531 :)